azure/services/agentpools/spec.go

```diff
@@ -201,6 +207,10 @@ func (s *AgentPoolSpec) Parameters(existing interface{}) (params interface{}, er
 	if s.VnetSubnetID != "" {
 		vnetSubnetID = &s.VnetSubnetID
 	}
+	var additionalTags infrav1.Tags
```

Comment:
I think the pattern you're seeing here is only for the fields that are a pointer in the Azure API types but are plain values in the CAPZ `AgentPoolSpec`. The type of some fields like `s.MaxPods` matches the Azure API types and those values are passed through without a similar check here.

Since `converters.TagsToMap()` handles that type conversion between `map[string]string` and `map[string]*string` already and your other PR fixed the `nil` map case, I think this code here could be removed and the value could be passed through directly in the return like:
```go
			Tags:                 converters.TagsToMap(s.AdditionalTags),
```